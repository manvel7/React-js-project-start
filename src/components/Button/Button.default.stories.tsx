import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Button, {IProps} from "./Button"

const meta: Meta<IProps> = {
  title: "Components/Button/Default",
  component: Button,
  argTypes: {
    onClick: {action: "onClick"}
  }
}

const ButtonStory: Story<IProps> = (args) => <Button {...args}>Button</Button>

const Large = ButtonStory.bind({})
Large.args = {
  size: "lg",
  variant: "default"
}

const LargeDisabled = ButtonStory.bind({})
LargeDisabled.args = {
  size: "lg",
  variant: "default",
  isDisabled: true
}

const Medium = ButtonStory.bind({})
Medium.args = {
  size: "md",
  variant: "default"
}

const MediumDisabled = ButtonStory.bind({})
MediumDisabled.args = {
  size: "md",
  variant: "default",
  isDisabled: true
}

const Small = ButtonStory.bind({})
Small.args = {
  size: "sm",
  variant: "default"
}

const SmallDisabled = ButtonStory.bind({})
SmallDisabled.args = {
  size: "sm",
  variant: "default",
  isDisabled: true
}

const Info = ButtonStory.bind({})
Info.args = {
  state: "info"
}

const Success = ButtonStory.bind({})
Success.args = {
  state: "success"
}

const Warn = ButtonStory.bind({})
Warn.args = {
  state: "warn"
}

const Error = ButtonStory.bind({})
Error.args = {
  state: "error"
}

export {
  Large,
  LargeDisabled,
  Medium,
  MediumDisabled,
  Small,
  SmallDisabled,
  Info,
  Success,
  Warn,
  Error
}

export default meta
