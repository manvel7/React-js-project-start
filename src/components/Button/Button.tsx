import {ButtonHTMLAttributes} from "react"
import styled, {css, StyledProps} from "styled-components"
import {h4Mixin, h6Mixin, smallMixin} from "../../styles/typography"

export type ButtonSize = "sm" | "md" | "lg"
export type ButtonVariant = "default" | "outlined" | "text"
export type ButtonState = "info" | "warn" | "success" | "error"

export interface IProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  isDisabled?: boolean
  size?: ButtonSize
  variant?: ButtonVariant
  state?: ButtonState
}

const lgMixin = css<StyledProps<IProps>>`
  border-radius: 0.8rem;
  padding: 0.8rem 1.6rem;
  ${h4Mixin<IProps>()}
`

const smMixin = css<StyledProps<IProps>>`
  border-radius: 0.3rem;
  padding: 0.2rem 1.6rem;
  ${smallMixin<IProps>()}
`

const mdMixin = css<StyledProps<IProps>>`
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  ${h6Mixin<IProps>()}
`

const outlinedMixin = css<StyledProps<IProps>>`
  background-color: transparent;
  border: 0.1rem solid
    ${({isDisabled, state, theme}) => {
      if (isDisabled) {
        return theme.palette.disabled
      }

      switch (state) {
        case "error":
          return theme.palette.alert.error
        case "warn":
          return theme.palette.alert.warn
        case "success":
          return theme.palette.alert.success
        case "info":
        default:
          return theme.palette.primary
      }
    }};
  box-sizing: border-box;
  color: ${({isDisabled, theme}) =>
    isDisabled ? theme.palette.text.darkDimmed : theme.palette.primary};
`

const textMixin = css<StyledProps<IProps>>`
  background-color: transparent;
  color: ${({isDisabled, theme}) =>
    isDisabled ? theme.palette.text.darkDimmed : theme.palette.primary};
`

const defaultMixin = css<StyledProps<IProps>>`
  background-color: ${({isDisabled, state, theme}) => {
    if (isDisabled) {
      return theme.palette.background.disabled
    }

    switch (state) {
      case "error":
        return theme.palette.alert.error
      case "warn":
        return theme.palette.alert.warn
      case "success":
        return theme.palette.alert.success
      case "info":
      default:
        return theme.palette.primary
    }
  }};
  color: ${({isDisabled, theme}) =>
    isDisabled ? theme.palette.text.darkDimmed : theme.palette.text.white};
`

const infoMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.info};
`

const warnMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.warn};
`

const errorMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.error};
`

const successMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.success};
`

const Button = styled.button<IProps>`
  border: 0.2rem solid transparent;
  cursor: pointer;

  ${({size}) => {
    switch (size) {
      case "lg":
        return lgMixin
      case "sm":
        return smMixin
      case "md":
      default:
        return mdMixin
    }
  }}

  ${({variant}) => {
    switch (variant) {
      case "outlined":
        return outlinedMixin
      case "text":
        return textMixin
      case "default":
      default:
        return defaultMixin
    }
  }}

  ${({state, variant}) => {
    if (!variant || variant === "default") {
      return ""
    }

    switch (state) {
      case "info":
        return infoMixin
      case "success":
        return successMixin
      case "warn":
        return warnMixin
      case "error":
        return errorMixin
      default:
        return ""
    }
  }}
`

export default Button
