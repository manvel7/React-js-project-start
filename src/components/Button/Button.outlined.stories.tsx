import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Button, {IProps} from "./Button"

const meta: Meta<IProps> = {
  title: "Components/Button/Outlined",
  component: Button,
  argTypes: {
    onClick: {action: "onClick"}
  }
}

const ButtonStory: Story<IProps> = (args) => <Button {...args}>Button</Button>

const Large = ButtonStory.bind({})
Large.args = {
  size: "lg",
  variant: "outlined"
}

const LargeDisabled = ButtonStory.bind({})
LargeDisabled.args = {
  size: "lg",
  variant: "outlined",
  isDisabled: true
}

const Medium = ButtonStory.bind({})
Medium.args = {
  size: "md",
  variant: "outlined"
}

const MediumDisabled = ButtonStory.bind({})
MediumDisabled.args = {
  size: "md",
  variant: "outlined",
  isDisabled: true
}

const Small = ButtonStory.bind({})
Small.args = {
  size: "sm",
  variant: "outlined"
}

const SmallDisabled = ButtonStory.bind({})
SmallDisabled.args = {
  size: "sm",
  variant: "outlined",
  isDisabled: true
}

const Info = ButtonStory.bind({})
Info.args = {
  variant: "outlined",
  state: "info"
}

const Success = ButtonStory.bind({})
Success.args = {
  variant: "outlined",
  state: "success"
}

const Warn = ButtonStory.bind({})
Warn.args = {
  variant: "outlined",
  state: "warn"
}

const Error = ButtonStory.bind({})
Error.args = {
  variant: "outlined",
  state: "error"
}

export {
  Large,
  LargeDisabled,
  Medium,
  MediumDisabled,
  Small,
  SmallDisabled,
  Info,
  Success,
  Warn,
  Error
}

export default meta
