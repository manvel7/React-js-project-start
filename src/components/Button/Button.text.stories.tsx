import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Button, {IProps} from "./Button"

const meta: Meta<IProps> = {
  title: "Components/Button/Text",
  component: Button,
  argTypes: {
    onClick: {action: "onClick"}
  }
}

const ButtonStory: Story<IProps> = (args) => <Button {...args}>Button</Button>

const Large = ButtonStory.bind({})
Large.args = {
  size: "lg",
  variant: "text"
}

const LargeDisabled = ButtonStory.bind({})
LargeDisabled.args = {
  size: "lg",
  variant: "text",
  isDisabled: true
}

const Medium = ButtonStory.bind({})
Medium.args = {
  size: "md",
  variant: "text"
}

const MediumDisabled = ButtonStory.bind({})
MediumDisabled.args = {
  size: "md",
  variant: "text",
  isDisabled: true
}

const Small = ButtonStory.bind({})
Small.args = {
  size: "sm",
  variant: "text"
}

const SmallDisabled = ButtonStory.bind({})
SmallDisabled.args = {
  size: "sm",
  variant: "text",
  isDisabled: true
}

const Info = ButtonStory.bind({})
Info.args = {
  variant: "text",
  state: "info"
}

const Success = ButtonStory.bind({})
Success.args = {
  variant: "text",
  state: "success"
}

const Warn = ButtonStory.bind({})
Warn.args = {
  variant: "text",
  state: "warn"
}

const Error = ButtonStory.bind({})
Error.args = {
  variant: "text",
  state: "error"
}

export {
  Large,
  LargeDisabled,
  Medium,
  MediumDisabled,
  Small,
  SmallDisabled,
  Info,
  Success,
  Warn,
  Error
}

export default meta
