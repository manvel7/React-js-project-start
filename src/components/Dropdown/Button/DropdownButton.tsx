import React, {FunctionComponent} from "react"
import {faCaretDown} from "@fortawesome/pro-solid-svg-icons"
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome"
import Button from "../../Button"
import ButtonGroup from "../../ButtonGroup"
import {IProps as IDropdownProps} from "../Dropdown"

interface IProps extends IDropdownProps {
  isOpened: boolean
  onToggleOpen: () => void
}

const DropdownButton: FunctionComponent<IProps> = ({
  size,
  onButtonClick,
  onToggleOpen,
  title,
  isOpened
}) => (
  <ButtonGroup>
    <Button size={size} onClick={onButtonClick}>
      {title}
    </Button>
    <Button size={size} onClick={onToggleOpen}>
      <FontAwesomeIcon
        icon={faCaretDown}
        size="sm"
        rotation={isOpened ? 180 : undefined}
      />
    </Button>
  </ButtonGroup>
)

export default DropdownButton
