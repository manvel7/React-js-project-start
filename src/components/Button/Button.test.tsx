import React from "react"
import {render, screen} from "@testing-library/react"
import {UiProvider} from "../../styles"
import theme from "../../styles/theme"
import {pxToRem} from "../../utils/unit"
import Button from "./Button"

describe("Button", () => {
  describe("Small 'sm' size", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button size="sm">Small Button</Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Button size="sm">Small Button</Button>
        </UiProvider>
      )

      const element = screen.getByRole("button")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule("border-radius", "0.3rem")
      expect(component).toHaveStyleRule(
        "font-size",
        pxToRem(
          theme.typography.variants.caption.size,
          theme.typography.htmlFontSize
        )
      )
    })
  })
  describe("Medium 'md' size", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button size="md">Medium Button</Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Button size="md">Medium Button</Button>
        </UiProvider>
      )

      const element = screen.getByRole("button")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule("border-radius", "0.4rem")
      expect(component).toHaveStyleRule(
        "font-size",
        pxToRem(
          theme.typography.variants.h6.size,
          theme.typography.htmlFontSize
        )
      )
    })
  })
  describe("Large 'lg' size", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button size="lg">Large Button</Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Button size="lg">Large Button</Button>
        </UiProvider>
      )

      const element = screen.getByRole("button")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule("border-radius", "0.8rem")
      expect(component).toHaveStyleRule(
        "font-size",
        pxToRem(
          theme.typography.variants.h4.size,
          theme.typography.htmlFontSize
        )
      )
    })
  })
  describe("Default variant", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="default">Default Button</Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Button variant="default">Default Button</Button>
        </UiProvider>
      )

      const element = screen.getByRole("button")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.primary
      )
      expect(component).toHaveStyleRule("color", theme.palette.text.white)
    })

    it("with state 'info' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="default" state="info">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("with state 'warn' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="default" state="warn">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("with state 'error' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="default" state="error">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("with state 'success' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="default" state="success">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
  describe("Default variant disabled", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="default" isDisabled={true}>
            Default Button disabled
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Button variant="default" isDisabled={true}>
            Default Button disabled
          </Button>
        </UiProvider>
      )

      const element = screen.getByRole("button")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.background.disabled
      )
      expect(component).toHaveStyleRule("color", theme.palette.text.darkDimmed)
    })
  })
  describe("Outlined variant", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="outlined">Outlined Button</Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Button variant="outlined">Outlined Button</Button>
        </UiProvider>
      )

      const element = screen.getByRole("button")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule("background-color", "transparent")
      expect(component).toHaveStyleRule("color", theme.palette.primary)
      expect(component).toHaveStyleRule("box-sizing", "border-box")
      expect(component).toHaveStyleRule(
        "border",
        `0.1rem solid ${theme.palette.primary}`
      )
    })

    it("with state 'info' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="outlined" state="info">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("with state 'warn' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="outlined" state="warn">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("with state 'error' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="outlined" state="error">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("with state 'success' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="outlined" state="success">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
  describe("Outlined variant disabled", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="outlined" isDisabled={true}>
            Outlined Button disabled
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Button variant="outlined" isDisabled={true}>
            Outlined Button disabled
          </Button>
        </UiProvider>
      )

      const element = screen.getByRole("button")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule("background-color", "transparent")
      expect(component).toHaveStyleRule("color", theme.palette.text.darkDimmed)
      expect(component).toHaveStyleRule("box-sizing", "border-box")
      expect(component).toHaveStyleRule(
        "border",
        `0.1rem solid ${theme.palette.disabled}`
      )
    })
  })
  describe("Text variant", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="text">Text Button</Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Button variant="text">Text Button</Button>
        </UiProvider>
      )

      const element = screen.getByRole("button")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule("background-color", "transparent")
      expect(component).toHaveStyleRule("color", theme.palette.primary)
    })

    it("with state 'info' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="text" state="info">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("with state 'warn' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="text" state="warn">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("with state 'error' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="text" state="error">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("with state 'success' should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="text" state="success">
            Default Info Button
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
  describe("Text variant disabled", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Button variant="text" isDisabled={true}>
            Text Button disabled
          </Button>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Button variant="text" isDisabled={true}>
            Text Button disabled
          </Button>
        </UiProvider>
      )

      const element = screen.getByRole("button")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule("background-color", "transparent")
      expect(component).toHaveStyleRule("color", theme.palette.text.darkDimmed)
    })
  })
})
