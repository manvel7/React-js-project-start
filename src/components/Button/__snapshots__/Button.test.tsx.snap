// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Default variant disabled should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: #E6E6E6;
  color: #808080;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Button disabled
</button>
`;

exports[`Button Default variant should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Button
</button>
`;

exports[`Button Default variant with state 'error' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: #CA0000;
  color: #FEFEFE;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Default variant with state 'info' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Default variant with state 'success' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: #0C5B0F;
  color: #FEFEFE;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Default variant with state 'warn' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: #D4A600;
  color: #FEFEFE;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Large 'lg' size should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.8rem;
  padding: 0.8rem 1.6rem;
  font-size: 2.2rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Large Button
</button>
`;

exports[`Button Medium 'md' size should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Medium Button
</button>
`;

exports[`Button Outlined variant disabled should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  border: 0.1rem solid #DDDDDD;
  box-sizing: border-box;
  color: #808080;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Outlined Button disabled
</button>
`;

exports[`Button Outlined variant should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  border: 0.1rem solid #001254;
  box-sizing: border-box;
  color: #001254;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Outlined Button
</button>
`;

exports[`Button Outlined variant with state 'error' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  border: 0.1rem solid #CA0000;
  box-sizing: border-box;
  color: #001254;
  color: #CA0000;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Outlined variant with state 'info' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  border: 0.1rem solid #001254;
  box-sizing: border-box;
  color: #001254;
  color: #001254;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Outlined variant with state 'success' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  border: 0.1rem solid #0C5B0F;
  box-sizing: border-box;
  color: #001254;
  color: #0C5B0F;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Outlined variant with state 'warn' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  border: 0.1rem solid #D4A600;
  box-sizing: border-box;
  color: #001254;
  color: #D4A600;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Small 'sm' size should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.3rem;
  padding: 0.2rem 1.6rem;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Small Button
</button>
`;

exports[`Button Text variant disabled should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  color: #808080;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Text Button disabled
</button>
`;

exports[`Button Text variant should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  color: #001254;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Text Button
</button>
`;

exports[`Button Text variant with state 'error' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  color: #001254;
  color: #CA0000;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Text variant with state 'info' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  color: #001254;
  color: #001254;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Text variant with state 'success' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  color: #001254;
  color: #0C5B0F;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;

exports[`Button Text variant with state 'warn' should match snapshot 1`] = `
.c0 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: transparent;
  color: #001254;
  color: #D4A600;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

<button
  class="c0"
>
  Default Info Button
</button>
`;
